docs/reference/cat/segments.asciidoc

```diff
@@ -1,73 +1,123 @@
-test  0     p      127.0.0.1 _0               0          1            0  3kb        2042 false     true       {lucene_version}   true
-test1 0     p      127.0.0.1 _0               0          1            0  3kb        2042 false     true       {lucene_version}   true
---------------------------------------------------
-// TESTRESPONSE[s/3kb/\\d+(\\.\\d+)?[mk]?b/ s/2042/\\d+/ non_json]
 
-The output shows information about index names and shard numbers in the first
-two columns.
+[[cat-segments-path-params]]
+==== {api-path-parms-title}
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=index]
+
+
+[[cat-segments-query-params]]
+==== {api-query-parms-title}
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=bytes]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=http-format]
+
+include::{docdir}/rest-api/common-parms.asciidoc[tag=cat-h]
++
+--
+If you do not specify which columns to include, the API returns the default
+columns in the order listed below. If you explicitly specify one or more
+columns, it only returns the specified columns.
+
+Valid columns are:
 
-If you only want to get information about segments in one particular index,
-you can add the index name in the URL, for example `/_cat/segments/test`. Also,
-several indexes can be queried like `/_cat/segments/test,test1`
+`index`, `i`, `idx`::
+(Default) Name of the index, such as `twitter`.
 
+`shard`, `s`, `sh`::
+(Default) Name of the shard.
 
-The following columns provide additional monitoring information:
+`prirep`, `p`, `pr`, `primaryOrReplica`::
+(Default) Shard type. Returned values are `primary` or `replica`.
 
-prirep::        Whether this segment belongs to a primary or replica shard.
+`ip`::
+(Default) IP address of the segment's shard, such as `127.0.1.1`.
 
-ip::            The ip address of the segment's shard.
+`segment`::
+(Default) Name of the segment, such as `_0`. The segment name is derived from
+the segment generation and used internally to create file names in the directory
+of the shard.
 
-segment::       A segment name, derived from the segment generation. The name
-                is internally used to generate the file names in the directory
-                of the shard this segment belongs to.
+`generation`::
+(Default) Generation number, such as `0`. {es} increments this generation number
+for each segment written. {es} then uses this number to derive the segment name.
 
-generation::    The generation number is incremented with each segment that is written.
-                The name of the segment is derived from this generation number.
+`docs.count`::
+(Default) Number of non-deleted documents in the segment, such as `25`. This
+number is based on Lucene documents and may include documents from
+<<nested,nested>> fields.
+                
+`docs.deleted`::
+(Default) Number of deleted documents in the segment, such as `0`. This number
+is based on Lucene documents. {es} reclaims the disk space of Lucene documents a
+segment is merged.
```

Comment:
I think this last sentence is missing a "when"? Not sure I'd repeat "Lucene" -- I think the more important point is that the disk space from _deleted_ documents is reclaimed when segments are merged.

Reply:
Thanks @debadair. Fixed with 96a1d9c858e.